public/app/plugins/panel/timeseries/TimeSeriesPanel.tsx

```diff
@@ -37,6 +37,21 @@ export const TimeSeriesPanel = ({
     return getFieldLinksForExplore({ field, rowIndex, splitOpenFn: onSplitOpen, range: timeRange });
   };
 
+  const { annotations, exemplars } = useMemo(() => {
+    let annotations = [];
+    let exemplars = [];
+
+    for (let frame of data.annotations ?? []) {
+      if (frame.name === 'exemplar') {
+        exemplars.push(frame);
+      } else {
+        annotations.push(frame);
+      }
+    }
+
+    return { annotations, exemplars };
```

Comment:
```suggestion
    return { 
       annotations: annotations.length ? annotations : undefined, 
       exemplars: exemplars.length ? exemplars : undefined,
    };
```
This will avoid creating the annotations+exemplars components when they do not exist (most of the time